### PR TITLE
Create datatypes.sql

### LINKED ISSUE #25: Override `dbt-core` datatype macros with Athena-specific implementations
`dbt-core` provides a number of macros to support cross-database compatibility for dbt packages: https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros

A few of the datatype macros have defaults which are incorrect for Athena, notably:
- `float`
- `numeric`
- `int`

### PR DESCRIPTION
**Pull request by owenprough-sift**
Closes #25

### CODE REVIEW

**Review by nicor88 — APPROVED** [review]

**Comment by Jrmyy** [comment]
Should we merge this in a `release/1.3.1` branch ?

**Comment by owenprough-sift** [comment]
> Should we merge this in a `release/1.3.1` branch ?

It can probably wait until release 1.4.0; I don't know of any compatibility issues that would be addressed by these changes. The goal here was more for future-proofing for compatibility with other packages.

**Comment by nicor88** [comment]
@Jrmyy and @owenprough-sift I prefer option 1, as it's smoother for contributors. Also using releases in github we have the changelog automatically generated by github.

**Comment by Jrmyy** [comment]
Fine by me, we can merge this then.

**Comment by jessedobbelaere** [comment]
This PR would replace the https://github.com/lalalilo/athena-utils/blob/main/macros/dbt_utils/cross_db_utils/datatypes.sql from the `lalalilo/athena-utils` package that adds Athena support for [dbt-utils](https://github.com/dbt-labs/dbt-utils). 
In the future, we could probably move more to the dbt-athena package?

**Comment by nicor88** [comment]
@jessedobbelaere nice idea, maybe you can propose the last suggestion in an issue, to avoid to forget that in a comment of a merged PR. All the macros [here](https://github.com/lalalilo/athena-utils/tree/main/macros/dbt_utils/cross_db_utils) are super helpfull.

**Comment by owenprough-sift** [comment]
> This PR would replace this file https://github.com/lalalilo/athena-utils/blob/main/macros/dbt_utils/cross_db_utils/datatypes.sql from the `lalalilo/athena-utils` package that adds Athena support for [dbt-utils](https://github.com/dbt-labs/dbt-utils). In the future, we could probably move more to the dbt-athena package?

I agree that `athena-utils` could likely be obsoleted by implementing the dbt-core-recommended macros in this adapter itself. And it makes a lot of sense to me to copy/paste the implementations from `athena-utils` where appropriate. No need to reinvent the wheel (and reduces behavior changes for folks migrating from that package).

**Comment by hiro-o918** [comment]
Can this PR  conflict with my PR?
I solved datatypes issues by adopter implementation.

https://github.com/dbt-athena/dbt-athena/pull/30/files#diff-786e4e9820eed125a3ea010af0b100bcf95e87aec14a18adbb84ecbcafc4e0a1

File: dbt/include/athena/macros/adapters/datatypes.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+{%- macro athena__type_float() -%}
+    DOUBLE
+{%- endmacro -%}
+
+{%- macro athena__type_numeric() -%}
+    DECIMAL(38,6)
+{%- endmacro -%}
+
+{%- macro athena_type_int() -%}
```
Comment (nicor88) [inline dbt/include/athena/macros/adapters/datatypes.sql:12]: Small change required, there is an underscore missing, should be `athena__type_int()` to have the overwrite working.

**Comment by nicor88** [comment]
@hiro-o918 your `AthenaColumn` handle type mapping, in the end the final goal of it and this pr is the same, but I would like still to have this one merged, as it overwrites the `default` adapter behavior.

Even if I prefer generally python over jinjia macros, in this case I believe that having a macro for the data types is better.

**Comment by hiro-o918** [comment]
OK :+1:
thank you for checking that.

**Comment by Jrmyy** [comment]
I also think we should keep this PR. But I think this PR is missing the string type that was handled and transformed to varchar with the correct length.

**Comment by nicor88** [comment]
@owenprough-sift could you address the comments then we can merge this one.

**Comment by nicor88** [comment]
@Jrmyy I agree with you we should move this file to another utils section, but if we do so we should move macros, let's tackle in another PR and merge this one. I create this issue to keep track of this folder re-structuring: https://github.com/dbt-athena/dbt-athena/issues/41

**Review by nicor88 — APPROVED** [review]

**Pull request merged**